Sources/RealDeviceMapLib/Controller/InstanceControllers/AutoInstanceController.swift

```diff
@@ -271,13 +309,108 @@ class AutoInstanceController: InstanceControllerProto {
     }
 
     func getTask(mysql: MySQL, uuid: String, username: String?, account: Account?, timestamp: UInt64) -> [String: Any] {
-
         switch type {
-        case .quest:
+        case .pokemon:
+            let hit = jumpyCache.get(id: self.name) ?? 0
+            if hit == 0 {
+                try? initJumpyCoords()
+                jumpyCache.set(id: self.name, value: 1)
+            }
+
+            Log.debug(message: "getTask() - jumpy started for \(self.name)")
+
+            let (_, min, sec) = secondsToHoursMinutesSeconds()
+            let curSecInHour = min*60 + sec
+            jumpyLock.lock()
+
+            // increment location
+            let loc: Int = currentDevicesMaxLocation
+
+            let newLoc = determineNextJumpyLocation(curTime: curSecInHour, curLocation: loc)
+            firstRun = false
+
+            Log.debug(message:
+                "getTask() jumpy - Instance: \(name) - oldLoc=\(loc) & newLoc=\(newLoc)/\(jumpyCoords.count / 2)")
+
+            var currentJumpyCoord: JumpyCoord = JumpyCoord(id: 1, coord: Coord(lat: 0.0, lon: 0.0), spawnSeconds: 0)
+            if jumpyCoords.indices.contains(newLoc) {
+                currentDevicesMaxLocation = newLoc
+                currentJumpyCoord = jumpyCoords[newLoc]
+            } else {
+                if jumpyCoords.indices.contains(0) {
+                    currentDevicesMaxLocation = 0
+                    currentJumpyCoord = jumpyCoords[0]
+                } else {
+                    currentDevicesMaxLocation = -1
+                }
+            }
+            // locationLock.unlock()
+
+            jumpyLock.unlock()
+
+            var task: [String: Any] = [
+                "action": "scan_pokemon", "lat": currentJumpyCoord.coord.lat, "lon": currentJumpyCoord.coord.lon,
+                "min_level": minLevel, "max_level": maxLevel
+            ]
+
+            if InstanceController.sendTaskForLureEncounter { task["lure_encounter"] = true }
+
+            return task
+        case .tth:
+            // get route like for findy, specify fence and use tth = null
+            // with each gettask, just increment to next point in list
+            // requery the route every ???? min, set with cache above
+            // run until data length == 0, then output a message to tell user done
+            // since we actually care about laptime, use that variable
+            findyLock.lock()
+
+            let hit = findyCache.get(id: self.name) ?? 0
```

Comment:
what's that cache used for? it contains one element with value 1
You are only using it for being sure that init for findy coords is only run once at starting instance?!

Reply:
cache is used in findy for requery of data.  if the init wasn't perfect as workers move around the area, they find more spawnpoints.  i set the time to requery in initialization of the cache, 5min.  some had said less would be better, but i think it should be a variable one can config.